tiledb/sm/filter/test/compile_all_filters_main.cc

```diff
@@ -30,6 +30,14 @@
 
 int main() {
   (void)sizeof(tiledb::sm::FilterCreate);
-  (void)&tiledb::sm::FilterCreate::deserialize;
+  std::shared_ptr<tiledb::sm::Filter> (*fn_enc)(
+      tiledb::sm::Deserializer&,
+      const tiledb::sm::EncryptionKey&,
+      const uint32_t) = tiledb::sm::FilterCreate::deserialize;
+  (void)fn_enc;
+  std::shared_ptr<tiledb::sm::Filter> (*fn)(
+      tiledb::sm::Deserializer&, const uint32_t) =
+      tiledb::sm::FilterCreate::deserialize;
+  (void)fn;
```

Comment:
You can obtain the address of an overloaded function by using `static_cast`. See case 6 here: [overloaded_address](https://en.cppreference.com/w/cpp/language/overloaded_address). This allows making a one-line replacement that more directly expresses the intent of the statement.
```
  (void)static_cast<shared_ptr<tiledb::sm::Filter>(*)(tiledb::sm::Deserializer&, uint32_t)>(tiledb::sm::FilterCreate::deserialize);
```
You only need to reference one of the deserialize functions. The one with the shorter signature forwards to the longer one, so it will be linked in with only a reference to the shorter one.